custom_components/octopus_energy/api_client/__init__.py

```diff
@@ -797,15 +866,58 @@ async def async_get_heat_pump_configuration_and_status(self, account_id: str, eu
       async with client.post(url, json=payload, headers=headers) as heat_pump_response:
         response = await self.__async_read_response__(heat_pump_response, url)
 
-        if (response is not None and "data" in response and "octoHeatPumpControllerConfiguration" in response["data"]  and "octoHeatPumpControllerStatus" in response["data"]):
+        if (response is not None 
+            and "data" in response 
+            and "octoHeatPumpControllerConfiguration" in response["data"] 
+            and "octoHeatPumpControllerStatus" in response["data"]
+            and "octoHeatPumpLivePerformance" in response["data"]
+            and "octoHeatPumpLifetimePerformance" in response["data"]):
           return HeatPumpResponse.parse_obj(response["data"])
         
       return None
     
     except TimeoutError:
       _LOGGER.warning(f'Failed to connect. Timeout of {self._timeout} exceeded.')
       raise TimeoutException()
+
+  async def async_get_heat_pump_time_ranged_performance(self, euid: str, start_at: datetime, end_at: datetime):
```

Comment:
issue: This doesn't seem to be used anywhere, so please remove.

I'm also not wanting to expose _everything_ in the OE API and try and only expose things that can be used to make a person's home smarter (either via an automation or via a HA feature like the energy dashboard). Because you're exposing the live and lifetime data, I think this is enough.